clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/actions/DetachTitusTasks.java

```diff
@@ -0,0 +1,165 @@
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.google.common.collect.Sets;
+import com.netflix.spinnaker.clouddriver.event.EventMetadata;
+import com.netflix.spinnaker.clouddriver.saga.SagaCommand;
+import com.netflix.spinnaker.clouddriver.saga.flow.SagaAction;
+import com.netflix.spinnaker.clouddriver.saga.models.Saga;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import com.netflix.spinnaker.clouddriver.titus.TitusClientProvider;
+import com.netflix.spinnaker.clouddriver.titus.client.TitusClient;
+import com.netflix.spinnaker.clouddriver.titus.client.model.Job;
+import com.netflix.spinnaker.clouddriver.titus.client.model.ResizeJobRequest;
+import com.netflix.spinnaker.clouddriver.titus.client.model.Task;
+import com.netflix.spinnaker.clouddriver.titus.client.model.TerminateTasksAndShrinkJobRequest;
+import com.netflix.spinnaker.clouddriver.titus.credentials.NetflixTitusCredentials;
+import com.netflix.spinnaker.clouddriver.titus.deploy.description.DetachTitusInstancesDescription;
+import com.netflix.spinnaker.kork.exceptions.UserException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+import lombok.Builder;
+import lombok.Value;
+import lombok.experimental.NonFinal;
+import org.jetbrains.annotations.NotNull;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DetachTitusTasks implements SagaAction<DetachTitusTasks.DetachTitusTasksCommand> {
+  private final AccountCredentialsProvider accountCredentialsProvider;
+  private final TitusClientProvider titusClientProvider;
+
+  @Autowired
+  public DetachTitusTasks(
+      AccountCredentialsProvider accountCredentialsProvider,
+      TitusClientProvider titusClientProvider) {
+    this.accountCredentialsProvider = accountCredentialsProvider;
+    this.titusClientProvider = titusClientProvider;
+  }
+
+  @NotNull
+  @Override
+  public Result apply(
+      @NotNull DetachTitusTasks.DetachTitusTasksCommand command, @NotNull Saga saga) {
+    saga.log(
+        "Detaching Titus Tasks %s:%s:%s:%s",
+        command.description.getAccount(),
+        command.description.getRegion(),
+        command.getDescription().getAsgName(),
+        command.description.getInstanceIds());
+
+    AccountCredentials accountCredentials =
+        accountCredentialsProvider.getCredentials(command.description.getAccount());
+
+    TitusClient titusClient =
+        titusClientProvider.getTitusClient(
+            (NetflixTitusCredentials) accountCredentials, command.description.getRegion());
+
+    Job job = titusClient.findJobByName(command.description.getAsgName(), true);
+    if (job == null) {
+      saga.log("Job not found");
```

Comment:
This was pre-existing behavior. 

Seems potentially suspect to "successfully" complete an operation yet not actually do the work.